### ISSUE TITLE
Missing detection of `__findFile` usages.

### ISSUE THREAD

**Issue by voronind-com** [nix-community/nixd#622]
**Describe the bug**
Right now nixd reports __findFile as unused even when there are `<>` references.

**Logs (Required)**

**Configuration**

If you have some custom configuration, please paste it here.

**To Reproduce**

**Expected behavior**
Using `<path>` references marks `__findFile` arg as used.

**Screenshots**
![image](https://github.com/user-attachments/assets/7c96733d-a347-4552-b6be-a284b9e27f95)

**Additional context**

**Comment by inclyc** [comment]
I think `__findFile` is internal to C++ nix implementation thus we should not expose this explicitly. CC @roberth @Mic92

**Comment by voronind-com** [comment]
> I think `__findFile` is internal to C++ nix implementation thus we should not expose this explicitly. CC @roberth @Mic92

Why not? Is it gonna hurt users with other implementations? Because now it hurts people with cppnix, which are the majority.

Also this might only mean that nixd could use nother parameter being the nix implementation used.

**Comment by inclyc** [comment]
Please note that [the language reference](https://nix.dev/manual/nix/2.25/language/constructs/lookup-path) only states that 

> A lookup path is an identifier with an optional path suffix that resolves to a [path value](https://nix.dev/manual/nix/2.25/language/types#type-path) if the identifier matches a search path entry in [builtins.nixPath](https://nix.dev/manual/nix/2.25/language/builtins#builtins-nixPath). The algorithm for lookup path resolution is described in the documentation on [builtins.findFile](https://nix.dev/manual/nix/2.25/language/builtins#builtins-findFile).

But it never said *`<>` expressions will be expanded to `__findPath`*. It just said: it is guaranteed that "the algorithm" is the same as `builtins.findFile`.

So I don't think it is correct to treat this syntax as a user of `__findPath`.

> Why not? Is it gonna hurt users with other implementations? Because now it hurts people with cppnix, which are the majority.

It is not a consideration about implementations, but about what is "exposed" from a language's specification.

**Comment by voronind-com** [comment]
Do you mean this is about usability vs correctness?

**Comment by Mic92** [comment]
This is the first time, I hear of `__findFile`. Why do user have to specify it explicitly?

**Comment by inclyc** [comment]
> This is the first time, I hear of `__findFile`. Why do user have to specify it explicitly and how is it useful?

Actually nix does allow overriding many base "internal implementation". For example,

```
a < b
```

will be expanded into 

```
__lessThan a b
```

And in fact users can *redefine* `__lessThan`, e.g.

```
❯ nix repl
Nix 2.24.10
Type :? for help.
nix-repl> __lessThan = x: y: x + y

nix-repl> 1 < 2
3
```

**Comment by Mic92** [comment]
Oh. We shouldn't users encourage to do that I suppose. It's ok for debugging tools to show these internals, but endusers should not write their code in this way.

**Comment by inclyc** [comment]
> Oh. We shouldn't users encourage to do that I suppose. It's ok for debugging tools to show these internals, but endusers should not write their code in this way. If it would help to resolve the issue, maybe it could be turned into a warning that an internal symbol is used?

Agreed. Hi @voronind-com, would you like to accept this solution? i.e. let nixd mark warnings at internal symbols usages.

**Comment by voronind-com** [comment]
> > Oh. We shouldn't users encourage to do that I suppose. It's ok for debugging tools to show these internals, but endusers should not write their code in this way. If it would help to resolve the issue, maybe it could be turned into a warning that an internal symbol is used?
> 
> Agreed. Hi @voronind-com, would you like to accept this solution? i.e. let nixd mark warnings at internal symbols usages.

I'm unsure. The `__findFile` is commonly used as an easy reference from the git root, i.e. <packages> translates to the `packages` dir.

I still vote that `__findFile` should be considered used (as it technically is) when there are `<>`. As well as `__lessThen` when `<` is used etc.

**Comment by Mic92** [comment]
Well probably fair. I am just not sure how hard to special case it is nice.

**Comment by DylanRJohnston** [comment]
Den uses this to provide some nice DevX around deeply nested aspects https://den.oeiuwq.com/guides/angle-brackets/
